File: Project2/AptMgmtPortal/Data/Entity/MaintenanceRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AptMgmtPortal.Entity
{
    public class MaintenanceRequest
    {
        [Key]
        public int MaintenanceRequestId { get; set; }
      
        public DateTime TimeOpened { get; set; }
        public DateTime? TimeClosed { get; set; }

        public int OpeningUserId { get; set; }
        public int? ClosingUserId { get; set; }

        public string MaintenanceRequestType { get; set; }
        public MaintenanceCloseReason? CloseReason { get; set; }

        public string OpenNotes { get; set; }
        public string ResolutionNotes { get; set; }
        public string InternalNotes { get; set; }
        public string UnitNumber { get; set; }
    }
}

File: Project2/AptMgmtClient/src/app/model/maintenance-request.ts
import { MaintenanceCloseReason } from 'src/enums/maintenance-close-reason';

export interface MaintenanceRequest {
  maintenanceRequestId: number;
  timeOpened: Date;
  timeClosed: Date | null;
  openedBy: string;
  closedBy: string | null;
  maintenanceRequestType: string;
  closeReason: MaintenanceCloseReason | null;
  openNotes: string;
  resolutionNotes: string | null;
  unitNumber: string;
}
